tests/unit/plugins/module_utils/_api/api/test_client.py

```diff
@@ -36,14 +36,11 @@
 from ansible_collections.community.docker.plugins.module_utils._api.api.client import APIClient
 from ansible_collections.community.docker.plugins.module_utils._api.constants import DEFAULT_DOCKER_API_VERSION
 from ansible_collections.community.docker.plugins.module_utils._api.utils.utils import convert_filters
+from ansible_collections.community.docker.tests.unit.compat import mock
```

Comment:
This cannot be imported here or in any of the following files, since this file is Apache licensed, while the imported module is GPLv3+. Same for the other files in tests/unit/plugins/module_utils/_api.

Reply:
That is a bit of a stringent interpretation of the GPL, but fine. We might want to consider re-licensing it, though. In my opinion, this code is too trivial to really benefit from being GPL-licensed, anyways.

Also, it doesn't appear that having non-GPL test code is even [allowed](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#id5). I think that rule should be changed, but that is currently the rule.

Reply:
> That is a bit of a stringent interpretation of the GPL, but fine.

That's basically the reason why Ansible plugins must be licensed under GPLv3+, since they import code that is licensed under GPLv3+.

> Also, it doesn't appear that having non-GPL test code is even [allowed](https://github.com/ansible-collections/overview/blob/main/collection_requirements.rst#id5). I think that rule should be changed, but that is currently the rule.

I think it's a grey area, since the statement "All other code must be under the GPL-3.0-or-later" is followed by "These plugins ...", which shows that this statement mainly applies to all other code **in `plugins/`**. This definitely should be clarified though.

Reply:
> That's basically the reason why Ansible plugins must be licensed under GPLv3+, since they import code that is licensed under GPLv3+.

From my understanding, the license of the code in the repo doesn't really matter, as long as it's GPL-compatible. The GPL only applies once the code is distributed or compiled (in the case of compiled langauges). This project is already distributed under `GPL-3.0-or-later`. See e.g. https://github.com/ansible/ansible-lint#licensing.

Reply:
Hmm, I'm no lawyer, so ... no idea :-) In any case, making sure that code importing GPLv3+ licensed code is itself GPLv3+ licensed code seems like a safeguard to me that will avoid potential legal problems that might or might not exist, so I'd rather stick to that :-)

Reply:
> Hmm, I'm no lawyer, so ... no idea :-) In any case, making sure that code importing GPLv3+ licensed code is itself GPLv3+ licensed code seems like a safeguard to me that will avoid potential legal problems that might or might not exist, so I'd rather stick to that :-)

Seems reasonable to me.